### PR TITLE
docs: Bluetooth: Audio: Add notes on the design and data placements

### PR DESCRIPTION
**Pull request by Thalley**
Add notes on how the LE Audio stack is designed and why data is sometimes placed and controlled by the stack or the application.

### CODE REVIEW

File: doc/connectivity/bluetooth/bluetooth-le-audio-arch.rst [hunk 1]
```diff
@@ -327,3 +344,48 @@ The CSIP implementation supports the following roles
 
 The API reference for media control can be found in
 :ref:`Bluetooth Coordinated Sets <bluetooth_coordinated_sets>`.
+
+Specification correctness and data location
+-------------------------------------------
+
+The implementations are designed to ensure specification compliance to extend possible.
```
Comment (kruithofa) [inline doc/connectivity/bluetooth/bluetooth-le-audio-arch.rst:351]:
```suggestion
The implementations are designed to ensure specification compliance to the largest extent possible.
```

**Review by kruithofa — CHANGES_REQUESTED** [review]
one small typo

File: doc/connectivity/bluetooth/bluetooth-le-audio-arch.rst (hunk 1 above)
Comment (Thalley) [reply doc/connectivity/bluetooth/bluetooth-le-audio-arch.rst]: Went with a slightly different phrasing

**Review by kruithofa — DISMISSED** [review]

**Review by larsgk — DISMISSED** [review]

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Review by larsgk — APPROVED** [review]

**Review by kruithofa — APPROVED** [review]

**Review by aescolar — APPROVED** [review]

**Pull request merged**